CloudScriptTypings update

JoshuaStrunk opened the pull request:
This is an update to focused on making the code in the newly added CloudScriptTypings target easy to read and understand. This should allow the code to be more easily translated into generating typings for the js and node sdk. 

In addition the way in which descriptions and summaries are inserted into the generated file has been tuned to be better suited to TS tools such as  atom-typscript.

pgilmorepf commented on targets/cloudscript-typings/make.js:
```diff
@@ -26,3 +28,122 @@ exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
+*/
+function MakeDatatype(datatype, api, sourceDir) {
+  var stringLiteralTemplate = compileTemplate(sourceDir,"StringLiteral");
+  var interfaceTemplate = compileTemplate(sourceDir,"Interface");
```
Calls to EJS are reasonably expensive (It loads and compiles an entire file)
I would not fetch both templates here but rather...

pgilmorepf commented on targets/cloudscript-typings/make.js:
```diff
@@ -26,3 +28,122 @@ exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
+    postColon += "[]";
+  }
+
+  return GetDescription(property.description, "   ") + "   "+ preColon + " : " + postColon + ",\n";
```
You've hard coded your tab-spacing here.
I'd add a parameter that makes the spacing an input parameter of GetProperty, the same way as GetDescription.

pgilmorepf commented on targets/cloudscript-typings/make.js:
```diff
@@ -26,3 +28,122 @@ exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
+    locals.enumvalues = datatype.enumvalues;
+    return  GetDescription(datatype.description, "")+ stringLiteralTemplate(locals);
+  }
+  else {
```
... and the datatype template here.

pgilmorepf commented on targets/cloudscript-typings/make.js:
```diff
@@ -26,3 +28,122 @@ exports.makeServerAPI = function (apis, sourceDir, apiOutputDir) {
+    name: datatype.name,
+  };
+
+  if(datatype.isenum) {
```
... fetch the enum template here...

pgilmorepf replied:
(Alternately, you could fetch the templates once, and store them as a global since you're going to call them so many times)

pgilmorepf commented:
Minor comments.  If you make the revisions, awesome.  If not, I'll wait a day or so for you to have time to read them, otherwise I'll accept it and make the revisions myself tomorrow.

JoshuaStrunk commented:
Thank you for the comments. I have changed my compileTemplate function to have a compiledTemplates property on which I cache compiled templates. This should solve the issue of needlessly recompiling the templates. I have also removed the hard coded tab-spacing.
